### PR TITLE
updates comments and changes resetEpoch parameters

### LINKED ISSUE #193: Epochs: Remove Epoch setup parameters from resetEpoch and add counter
### Issue To Be Solved
EpochSetup parameters are not used by reset Epoch, so they can be removed from the function parameter list. We also need to include the current epoch counter.

### Suggest A Solution
* Remove `clusterQCs` and  `dkgKeys`
* Set the epoch metadata versions of those to `nil`
* Add `currentEpochCounter` as a parameter to `resetEpoch` and revert if it doesn't match the current epoch counter stored by the smart contract.
* Fix the reset epoch transaction and tests to use these changes

cc @jordanschalm

### PR DESCRIPTION
**Pull request by joshuahannan**
Closes https://github.com/onflow/flow/issues/441
Closes #193 

When going through the epoch docs and FLIPs, I also read through the comments in the Epoch smart contracts and updated/added a lot of them

Also, I removed some of the unused parameters in the `resetEpoch` function and added the `currentEpochCounter` parameter

### CODE REVIEW

File: transactions/epoch/admin/reset_epoch.cdc [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
-transaction(randomSource: String,
+transaction(currentEpochCounter: UInt64,
+            randomSource: String,
             newPayout: UFix64,
             startView: UInt64,
             endView: UInt64,
-            collectorClusters: [String]
-            clusterQCs: [String],
-            dkgPubKeys: [String]) {
+            collectorClusters: [String]) {
```
Comment (jordanschalm) [inline transactions/epoch/admin/reset_epoch.cdc:9]: If it is safe from the smart contract perspective to omit the `collectorClusters` input as well I think we should.

File: contracts/epochs/FlowEpoch.cdc [hunk 2]
```diff
@@ -402,8 +421,8 @@ pub contract FlowEpoch {
                     stakingEndView: startView + FlowEpoch.configurableMetadata.numViewsInStakingAuction - (1 as UInt64),
                     totalRewards: FlowIDTableStaking.getEpochTokenPayout(),
                     collectorClusters: collectorClusters,
-                    clusterQCs: clusterQCs,
-                    dkgKeys: dkgPubKeys)
+                    clusterQCs: [],
+                    dkgKeys: [])
```
[the remark this answers is not in the document]
Comment (joshuahannan) [reply contracts/epochs/FlowEpoch.cdc:425]: So we would still include these as parameters to the `resetEpoch` function, but would just make them empty arrays in the transaction? Isn't that already what we were doing before?

[the remark this answers is not in the document]
Comment (joshuahannan) [reply contracts/epochs/FlowEpoch.cdc:425]: Oh, I see. We'd remove it from the reset_epoch transaction parameter list, but keep it in the `resetEpoch()` function parameter list. Got it. 👍

**Comment by joshuahannan** [comment]
I addressed all your comments in 2af7601  @jordanschalm.
Thanks for the quick feedback! 😃

**Review by jordanschalm — APPROVED** [review]
🎸

**Pull request merged**